### ISSUE TITLE
pes.py

### ISSUE THREAD

**Issue by dnjohnstone** [hyperspy/hyperspy#979]
What is pes.py doing? It's a file entirely commented - I guess someone started it ages ago and it's just hanging around, but it seems a bit odd to include it like this?

**Comment by dnjohnstone** [comment]
@francisco-dlp are we just removing this?

**Comment by francisco-dlp** [comment]
Sounds sensible.

**Comment by dnjohnstone** [comment]
Actually, I eat my own words maybe not so sensible... this is not too far away from being very useful - I just did a bit of digging and I think I'd rather try to reinstate this properly... I hadn't realised that PES was in this case meaning what I would have called XPS.

At this point we have:

1. A very old implementation of Shirley background subtraction
2. A model component pes_see.py for secondary electron emission

In principle, I think both of these are very useful tools and XPS is something we could definitely support in very positive ways imagining one long term facet of HyperSpy as a comprehensive suite for materials characterization. Mapping in XPS is also becoming much more common with new instruments (e.g. http://xpssimplified.com/imaging_mapping.php). I note the following in particular, which perhaps would provide a roadmap for doing this:

1. PES should probably be renamed to XPS

2. The Shirley background subtraction implementation could be reinstated and is perhaps rather more general than XPS only where it happens to be a standard as here (http://www.casaxps.com/help_manual/manual_updates/peak_fitting_in_xps.pdf)

3. The model fitting problem is very well posed in terms of physical constraints i.e. intensity ratios are physically rather tightly fixed (in short it should be constrained in very similar ways to our EDS model).

I should say that in my own experience of XPS some of the very newest software from Fisher is actually very good - but I think we are ahead on dealing with mapping data a bit similar to the early days in EM. In short, I think this might be a very good signal to add and there is much physical similarity with EDS signals.

I have some old XPS data of my own that I have good fits to and could make available to someone - I'm not sure I've got time to implement this myself right now but I do think we should find a way to do it.

**Comment by francisco-dlp** [comment]
These features are from HyperSpy's ancestor, EELSLab. I added them in order to analyse the data in [this paper](http://www.sciencedirect.com/science/article/pii/S0039602810002529). However, some (like the Shirley bg) got broken when refurbishing the code to create the more general HyperSpy and I never had the time to work on getting these features back.

I think that it is good to remove the commented out code from the repository. In parallel we could open a new issue requesting to get the feature back, and copy and paste the old and ugly code there.

**Comment by dnjohnstone** [comment]
Oh that's a nice example of exactly what came to my mind for a good use of this -- it's almost like you've done this before!

It would be really awesome if that kind of analysis were available with some specialised tools in HyperSpy, your plan sounds good - do you want to do that yourself? Or do you want me to do it?

**Comment by francisco-dlp** [comment]
Reimplementing shirley background subtraction sounds good, although it'll probably be worth checking if there is a well established implementation in another Python package.
